src/main/java/kr/hhplus/be/server/service/RestaurantService.java

```diff
@@ -0,0 +1,30 @@
+package kr.hhplus.be.server.service;
+
+import kr.hhplus.be.server.domain.restaurant.Restaurant;
+import kr.hhplus.be.server.domain.restaurant.RestaurantRepository;
+import kr.hhplus.be.server.dto.RestaurantSearchRequest;
+import kr.hhplus.be.server.infrastructure.external.NaverApiClient;
+
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class RestaurantService {
+
+    private final NaverApiClient naverApiClient;
+    private final RestaurantRepository repository;
+
+    public RestaurantService(NaverApiClient naverApiClient, RestaurantRepository repository) {
+        this.naverApiClient = naverApiClient;
+        this.repository = repository;
+    }
+
+    public List<Restaurant> searchRestaurants(RestaurantSearchRequest request) {
+        List<Restaurant> externalApiResponse = naverApiClient.search(request);
+        // 검색 기록 저장
+        repository.save(externalApiResponse);
+
```

Comment:
```suggestion
        keywordRepository.increaseCount(request.getQuery());
```
이런식으로 인기 키워드 카운트 증가 로직도 추가되어야 할거 같네요!